### PR TITLE
Update master to v0.21.0

### PR DESCRIPTION
**Pull request by raoulcollenteur**
# Short Description
Update master to v0.21.0

### CODE REVIEW

File: pastas/plots.py [hunk 1]
```diff
@@ -619,14 +627,18 @@ def __init__(self, ml, tmin=None, tmax=None, update_iter=None):
 
         # calculate RMSE residuals
         res = self._residuals(self.ml.parameters.initial.values)
-        r_rmse = np.sqrt(np.sum(res ** 2))
+        r_rmse = np.sqrt(np.mean(res ** 2))
```
Comment (martinvonk) [inline pastas/plots.py:630]: Not better to call pastas.stats.metrics.rmse(res=res) ?

**Review by martinvonk — COMMENTED** [review]
One comment on calculating the RMSE (4 times) seperate of `pastas.stats.metrics.rmse()` in plots.py

File: pastas/plots.py (hunk 1 above)
Comment (raoulcollenteur) [reply pastas/plots.py:630]: Agreed, but let's do that in the next version. Can you create a GH Issue for this?

Comment (dbrakenhoff) [reply pastas/plots.py:630]: This predates the changes to the metrics, they used to be calculated in the model methods, which could not be accessed with the callback in solve. But there is no reason not to change it now, so good idea to open a new issue.

File: pastas/plots.py [hunk 2]
```diff
@@ -561,27 +561,35 @@ class TrackSolve:
...
+    Calculated parameters per iteration are stored in a pandas.DataFrame::
+
+        track.parameters
 
     Other stored statistics include `track.evp` (explained variance
     percentage), `track.rmse_res` (root-mean-squared error of the residuals),
     `track.rmse_noise` (root mean squared error of the noise, only if
     noise=True).
 
     To interactively plot model optimiztion progress while solving pass
```
Comment (dbrakenhoff) [inline pastas/plots.py:589]: typo

File: pastas/stressmodels.py [hunk 3]
```diff
@@ -892,6 +892,54 @@ def to_dict(self, series=True):
         }
         return data
 
+    def variance_gain(self, model, istress=None):
```
Comment (dbrakenhoff) [inline pastas/stressmodels.py:895]: Good idea to add r as kwarg here, allows calculating uncertainty as function of distance. Note to self to open issue later, not requisite for this PR.

**Review by martinvonk — APPROVED** [review]
Created issue #402 to deal with the metrics calculation in plots.py.

Can be merged imo.

_edit_: Obviously @dbrakenhoff still has some comments :)

File: pastas/timer.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+class SolveTimer(tqdm):
+    """Progress indicator for model optimization.
+
+    Usage
+    ----- 
+    Print timer and number of iterations in console while running
+    `ml.solve()`::
+
+    >>> with SolveTimer() as t:
```
Comment (dbrakenhoff) [inline pastas/timer.py:15]: remove >>>?

File: pastas/timer.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+try:
+    from tqdm.auto import tqdm
+except ModuleNotFoundError:
+    raise ModuleNotFoundError("SolveTimer requires 'tqdm' to be installed.")
+
+
+class SolveTimer(tqdm):
```
Comment (dbrakenhoff) [inline pastas/timer.py:7]: I added a solver timeout argument that I apparently haven't pushed yet, I'll quickly add that in.

**Review by dbrakenhoff — COMMENTED** [review]
I'm quickly gonna add the maxtime argument to the SolveTimer(), after that I will approve :+1:

**Review by dbrakenhoff — APPROVED** [review]

**Review by martinvonk — APPROVED** [review]

**Pull request merged**